module/zfs/zio.c

```diff
@@ -1771,6 +1771,20 @@ zio_write_compress(zio_t *zio)
 		    zio->io_abd, NULL, lsize, zp->zp_complevel);
 		if (psize == 0 || psize >= lsize)
 			compress = ZIO_COMPRESS_OFF;
+	} else if (zio->io_flags & ZIO_FLAG_RAW_COMPRESS) {
+		size_t rounded = MIN((size_t)roundup(psize,
+		    spa->spa_min_alloc), lsize);
+
+		if (rounded != psize) {
+			abd_t *cdata = abd_get_from_buf(zio_buf_alloc(rounded),
+			    rounded);
+			abd_take_ownership_of_buf(cdata, B_TRUE);
```

Comment:
I think this is equivalent to
```suggestion
			abd_t *cdata = abd_alloc_linear(rounded, B_TRUE);
```